example/shared/src/commonMain/kotlin/com/splendo/kaluga/example/shared/viewmodel/keyboard/KeyboardViewModel.kt

```diff
@@ -17,21 +17,37 @@
 
 package com.splendo.kaluga.example.shared.viewmodel.keyboard
 
+import com.splendo.kaluga.architecture.observable.toUninitializedSubject
 import com.splendo.kaluga.architecture.viewmodel.BaseLifecycleViewModel
 import com.splendo.kaluga.example.shared.stylable.ButtonStyles
 import com.splendo.kaluga.keyboard.BaseKeyboardManager
 import com.splendo.kaluga.keyboard.FocusHandler
 import com.splendo.kaluga.resources.localized
 import com.splendo.kaluga.resources.view.KalugaButton
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
 
-class KeyboardViewModel<FH : FocusHandler>(keyboardManagerBuilder: BaseKeyboardManager.Builder<FH>, private val editFieldFocusHandler: FH) : BaseLifecycleViewModel(
+class KeyboardViewModel<FH : FocusHandler>(keyboardManagerBuilder: BaseKeyboardManager.Builder<FH>) : BaseLifecycleViewModel(
     keyboardManagerBuilder,
 ) {
 
+    private val _editFieldFocusHandler = MutableStateFlow<FH?>(null)
+    val editFieldFocusHandler = _editFieldFocusHandler.toUninitializedSubject(coroutineScope)
+
     private val keyboardManager: BaseKeyboardManager<FH> = keyboardManagerBuilder.create(coroutineScope)
 
+    init {
+        coroutineScope.launch {
+            _editFieldFocusHandler.filterNotNull().collect {
+                keyboardManager.show(it)
+            }
+        }
+    }
```

Comment:
If I remember correctly for SwiftUI and Compose this is not need, giving focus will auto display the keyboard. Not sure in XML and old iOS ViewControler thingy.

Reply:
This isnt called when a view is being focussed on though. This is called when a view is known to the ViewModel so the ViewModel can request focus.

This is how you should do show keyboard on init